packages/discovery-react-components/src/components/StructuredQuery/__tests__/StructuredQuery.test.tsx

```diff
@@ -1,16 +1,76 @@
 import React from 'react';
-import { render, RenderResult } from '@testing-library/react';
-import StructuredQuery from '../StructuredQuery';
+import { render, RenderResult, fireEvent } from '@testing-library/react';
+import StructuredQuery, { StructuredQueryProps } from '../StructuredQuery';
+import { wrapWithContext } from 'utils/testingUtils';
+import {
+  SearchContextIFC,
+  searchResponseStoreDefaults,
+  fetchDocumentsResponseStoreDefaults,
+  SearchApiIFC
+} from 'components/DiscoverySearch/DiscoverySearch';
+import DiscoveryV2 from 'ibm-watson/discovery/v2';
+import '@testing-library/jest-dom/extend-expect';
+
+interface Setup {
+  structuredQuery: RenderResult;
+}
+
+function setup(
+  {
+    projectFields,
+    fieldsStoreLoadingState = false,
+    fieldsStoreErrorState = false,
+    apiOverrides = {}
+  }: {
+    projectFields?: DiscoveryV2.Field[];
+    fieldsStoreLoadingState?: boolean;
+    fieldsStoreErrorState?: boolean;
+    apiOverrides?: Partial<SearchApiIFC>;
+  },
+  componentProps: Partial<StructuredQueryProps> = {}
+): Setup {
+  let context: Partial<SearchContextIFC> = {
+    searchResponseStore: {
+      ...searchResponseStoreDefaults
+    },
+    fetchDocumentsResponseStore: {
+      ...fetchDocumentsResponseStoreDefaults
+    },
+    fieldsStore: {
+      data: { fields: projectFields },
+      isLoading: fieldsStoreLoadingState,
+      isError: fieldsStoreErrorState,
+      parameters: {
+        projectId: ''
+      }
+    }
+  };
+  const structuredQuery = render(
+    wrapWithContext(<StructuredQuery {...componentProps} />, apiOverrides, context)
+  );
+  return {
+    structuredQuery
+  };
+}
 
 describe('<StructuredQuery />', () => {
+  let structuredQuery: RenderResult;
+  let projectFields: DiscoveryV2.Field[] | undefined = undefined;
+  let fieldsStoreLoadingState: boolean | undefined;
```

Comment:
shouldn't this just be a `boolean`? im not sure why it would ever be `undefined`

Reply:
I previously was following the pattern of resetting my component params to `undefined` after each test to bring us back to the same `undefined` state after each test. But you're right that it probably makes sense to set this to a default value of `false` here